### DISCUSSION TITLE
How to run this project locally so that i can try and contribute different themes?

### DISCUSSION THREAD

**Discussion by bhaveek424** [anuraghazra/github-readme-stats#1930] [Q&A] [answered]
So I want to know how can get this repo up and running on my local machine so that I can try different color schemes and contribute it.
After forking the repo, i ran `npm i ` and then tried the scripts `npm run preview-theme` but that throws an error 
```ts
Error: Parameter token or opts.auth is required
```

**Comment by rickstaa** [comment, marked as answer]
@bhaveek424 Thanks a lot for considering contributing to the GRS repository! 🎉 A local development guide can be found in the [CONTRIBUTING.md](https://github.com/anuraghazra/github-readme-stats/blob/master/CONTRIBUTING.md#local-development) file:

> To run & test github-readme-stats, you need to follow a few simple steps:-
_(make sure you already have a [Vercel](https://vercel.com/) account)_
>
> 1.  Install [Vercel CLI](https://vercel.com/download).
> 2.  Fork the repository and clone the code to your local machine.
> 3.  Run `npm install` in the repository root.
> 4.  Run the command "vercel" in the root and follow the steps there.
> 5.  Open `vercel.json` and set the maxDuration to 10.
> 6.  Create a `.env` file in the root of the directory.
> 7.  In the .env file add a new variable named "PAT_1" with your [GitHub Personal Access Token](https://docs.github.com/en/github/authenticating-to-github/creating-a-personal-access-token).
> 8.  Run the command "vercel dev" to start a development server at <https://localhost:3000>.

Please be aware that the [preview-theme.js](https://github.com/anuraghazra/github-readme-stats/blob/master/scripts/preview-theme.js) is not meant for local development. It is used in the [preview-theme.yml](https://github.com/anuraghazra/github-readme-stats/blob/master/.github/workflows/preview-theme.yml) GitHub action. This action runs when you create a PR with a new theme to show the theme in an issue comment (unfortunately, this action is broken at the time of writing see #1899). 

The recommended way to test out your themes is by following the steps in the contribution guidelines, adding your theme and using it through the theme parameter (i.e. `&theme=THEME_NAME`).  Let me know if you have any other questions. 🚀

**Reply by bhaveek424** [reply]
@rickstaa Thank you for answering my doubts. I'm gonna try doing what you explained. :)

**Reply by rickstaa** [reply]
Great! Could you please accept my answer if it works so we can label this question as answered 🤓.
